mathics/core/evaluation.py

```diff
@@ -144,6 +144,10 @@ def __init__(
         self.quiet_all = False
         self.recursion_depth = 0
 
+        # iteration is used to keep track of evaluation chains and is
+        # compared against $IterationLimit.
+        self.iteration_count = 0
```

Comment:
In MathicsSession, the same evaluation object is used in all the evaluations of the session. At which point `self.iteration_count` is reset to 0?

Reply:
Ok - thanks. So we should reset this on each test.

We've added a number of other fields too, so those should probably be reset as well.

Reply:
In the old Mathics, one `Evaluation` object was created each time a line in the CLI or a cell in the web interface was executed. At some point, we change this behavior because most of the time, the Evaluation object does not change between evaluations. Now, since the evaluation always starts with a call to `evaluation` in the MathicsSession object, it is sufficient to reset the value there.